src/pages/index.js

```diff
@@ -1,205 +1,265 @@
-const addCardModal = document.querySelector("#add-card-modal");
-
-const cardTitleInput = document.querySelector("#add-title-input");
-const cardUrlInput = document.querySelector("#add-url-input");
-
-const addCardForm = addCardModal.querySelector("#add-card-form");
-
-// Preview Modal
-const previewModal = document.querySelector("#preview-modal");
-const previewModalImageEl = previewModal.querySelector(".modal__image");
-const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
-
-/**************************************************************************
- *                                CARD.JS                                 *
+ *                            RENDERING CARDS                             *
  **************************************************************************/
 
 // Function to handle image click
 function handleImageClick(data) {
-    openModal(previewModal);
-    previewModalImageEl.src = data.link;
-    previewModalImageEl.alt = data.name;
-    previewModalCaptionEl.textContent = data.name;
+  previewPopup.open(data.name, data.link);
 }
 
-// Function to render a card
-//function renderCard(data) {
-    //const card = new Card (data, cardSelector, handleImageClick);
-    //cardListEl.prepend(card.getView());
-//}//
-
-function createCard (data) {
-    const card = new Card (data, cardSelector, handleImageClick);
-    return card.getView();
-}
+const cardSection = new Section(
+  {
+    renderer: (data) => {
+      const card = renderCard(data);
+      cardSection.addItem(card);
+    },
+  },
+  ".cards__list"
+);
 
-function renderCard (data) {
-    const cardElement = createCard(data);
-    cardListEl.prepend(cardElement);
+// Function to render a card
+function renderCard(data) {
+  const card = new Card(
+    data,
+    cardSelector,
+    () => handleImageClick(data),
+    (cardId) => handleDeleteCard(card, cardId),
+    (card, cardId, isLiked) => handleLikeToggle(card, cardId, isLiked)
+  );
+  return card.getView();
 }
 
-
-
-
-// Render initial cards
-initialCards.forEach(renderCard)
-
 /**************************************************************************
  *                               VALIDATION                               *
  **************************************************************************/
 
-const validationConfig = {
-    formSelector: ".modal__form",
-    inputSelector: ".modal__input",
-    submitButtonSelector: ".modal__button",
-    inactiveButtonClass: "modal__button_disabled",
-    inputErrorClass: "modal__input_type_error",
-    errorClass: "modal__error_visible"
-};
-
-const profileFormValidator = new FormValidator(validationConfig, profileEditForm);
+const profileFormValidator = new FormValidator(
+  validationConfig,
+  profileEditForm
+);
 const cardFormValidator = new FormValidator(validationConfig, addCardForm);
+const editAvatarValidator = new FormValidator(validationConfig, editAvatarForm);
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```